chemprop/args.py

```diff
@@ -383,6 +385,22 @@ class TrainArgs(CommonArgs):
     """Overwrites the default atom descriptors with the new ones instead of concatenating them"""
     no_bond_features_scaling: bool = False
     """Turn off atom feature scaling."""
+    frzn_mpnn: bool = False
+    """
+    Overwrites mpnn weights with weights from checkpoint model (specified checkpoint_frzn) and freezes them.
+    """
+    frzn_ffn_layers: int = 0
+    """
+    Overwrites weights for the first n layers of the ffn from checkpoint model (specified checkpoint_frzn), 
+    where n is specified in the input.
+    Automatically also freezes mpnn weights. 
+    """
+    freeze_first_only: bool = False
+    """
+    Determines whether or not to use checkpoint_frzn for just the first encoder.
+    Default (False) is to use the checkpoint to freeze all encoders.
+    (only relevant for number_of_molecules > 1, where checkpoint model has number_of_molecules = 1)
+    """
```

Comment:
For the default behavior where it freezes all encoders, this is supposing distinct mpnns for the two molecules or is it one mpnn copied for both?

Reply:
If the checkpoint file contains two molecules, then the two different MPNNs will be used and frozen. If you are training a model for two molecules, but your frozen checkpoint only includes one molecule, then it will, by default, copy the single MPNN and use it for both molecules and freeze it (unless you select freeze first only, in which case the MPNN will only be used for the first molecule).

Reply:
That's a great way to handle that